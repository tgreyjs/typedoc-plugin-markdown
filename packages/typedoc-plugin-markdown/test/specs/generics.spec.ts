import * as Handlebars from 'handlebars';
import { SignatureReflection } from 'typedoc';

import { TestApp } from '../test-app';

describe(`Generics:`, () => {
  let testApp: TestApp;
  let partial: Handlebars.TemplateDelegate;
  let reflectionTemplate: Handlebars.TemplateDelegate;

  beforeAll(() => {
    testApp = new TestApp(['generics.ts']);
    partial = TestApp.getPartial('member.signature');
  });

  beforeEach(async () => {
    await testApp.bootstrap();
    TestApp.stubPartials(['comment', 'member.signature', 'members']);
    TestApp.stubHelpers(['toc', 'breadcrumbs', 'hierarchy', 'returns']);
    reflectionTemplate = TestApp.getTemplate('reflection');
  });

  test(`should compile class with type params`, () => {
    expect(
      TestApp.compileTemplate(reflectionTemplate, {
        model: testApp.findReflection('ClassWithTypeParams'),
        project: testApp.project,
      }),
    ).toMatchSnapshot();
  });

  test(`should compile function with a simple type param'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('functionWithTypeParam')
          .signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile function with complex type params'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('functionWithTypeParams')
          .signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });
});
