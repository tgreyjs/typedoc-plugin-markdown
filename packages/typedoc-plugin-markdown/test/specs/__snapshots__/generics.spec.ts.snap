// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generics: should compile class with type params 1`] = `
"[helper: breadcrumbs]

# Class: ClassWithTypeParams<T, V\\\\>

[partial: comment]

## Type parameters

| Name | Description |
| :------ | :------ |
| \`T\` | Some type param |
| \`V\` | Some other type param |

[helper: toc]

[partial: members]
"
`;

exports[`Generics: should compile function with a simple type param' 1`] = `
"▸ \`Const\` **functionWithTypeParam**<A\\\\>(): \`boolean\`

[partial: comment]

##### Type parameters

| Name |
| :------ |
| \`A\` |

##### Returns

\`boolean\`
"
`;

exports[`Generics: should compile function with complex type params' 1`] = `
"▸ \`Const\` **functionWithTypeParams**<A, B, C\\\\>(): \`boolean\`

[partial: comment]

##### Type parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| \`A\` | \`A\`: [ClassWithTypeParams](classwithtypeparams.md)<string, number, A\\\\> | Comment for type \`A\` |
| \`B\` | \`B\` = \`string\` \\\\| \`boolean\` | Comment for type \`B\` |
| \`C\` | \`C\` = \`string\` | - |

##### Returns

\`boolean\`
"
`;
